src/app/redux/orderInputSlice.ts

```diff
@@ -0,0 +1,397 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+import * as adex from "alphadex-sdk-js";
+import { RootState } from "./store";
+import { createSelector } from "@reduxjs/toolkit";
+import { getRdt, RDT } from "../subscriptions";
+import { fetchBalances } from "./pairSelectorSlice";
+import { SdkResult } from "alphadex-sdk-js/lib/models/sdk-result";
+
+export enum OrderTab {
+  MARKET,
+  LIMIT,
+}
+
+export const PLATFORM_BADGE_ID = 1; //TODO: Get this data from the platform badge
+export const PLATFORM_FEE = 0.001; //TODO: Get this data from the platform badge
+
+export const OrderSide = adex.OrderSide;
+export type OrderSide = adex.OrderSide;
+export type Quote = adex.Quote;
+
+export interface OrderInputState {
+  token1Selected: boolean;
+  tab: OrderTab;
+  preventImmediateExecution: boolean;
+  side: OrderSide;
+  size: number;
+  minSize: number;
+  price: number;
+  toAmountEstimate: number;
+  slippage: number;
+  quote?: Quote;
+  description?: string;
+  transactionInProgress: boolean;
+  transactionResult?: string;
+}
+
+function adexOrderType(state: OrderInputState): adex.OrderType {
+  if (state.tab === OrderTab.MARKET) {
+    return adex.OrderType.MARKET;
+  }
+  if (state.tab === OrderTab.LIMIT) {
+    if (state.preventImmediateExecution) {
+      return adex.OrderType.POSTONLY;
+    } else {
+      return adex.OrderType.LIMIT;
+    }
+  }
+
+  throw new Error("Invalid order type");
+}
+
+const initialState: OrderInputState = {
+  token1Selected: true,
+  tab: OrderTab.MARKET,
+  preventImmediateExecution: false,
+  side: adex.OrderSide.BUY,
+  size: 0,
+  minSize: 0,
+  price: 0,
+  slippage: 0.01,
+  toAmountEstimate: 0,
+  transactionInProgress: false,
+};
+
+export const fetchQuote = createAsyncThunk<
+  Quote, // Return type of the payload creator
+  undefined, // set to undefined if the thunk doesn't expect any arguments
+  { state: RootState }
+>("orderInput/fetchQuote", async (_arg, thunkAPI) => {
+  const state = thunkAPI.getState();
+  if (state.pairSelector.address === "") {
+    throw new Error("Pair address is not initilized yet.");
+  }
+
+  let priceToSend = undefined;
+  let slippageToSend = undefined;
+
+  if (state.orderInput.tab === OrderTab.LIMIT) {
+    if (state.orderInput.price < 0) {
+      throw new Error("Invalid price");
+    }
+    priceToSend = state.orderInput.price;
+  } else {
+    slippageToSend = state.orderInput.slippage;
+  }
+
+  const response = await adex.getExchangeOrderQuote(
+    state.pairSelector.address,
+    adexOrderType(state.orderInput),
+    state.orderInput.side,
+    getSelectedToken(state).address,
+    state.orderInput.size,
+    PLATFORM_FEE,
+    priceToSend,
+    slippageToSend
+  );
+  const quote = JSON.parse(JSON.stringify(response.data));
+  return { ...quote };
+});
+
+export const setSizePercent = createAsyncThunk<
```

Comment:
Ok, fixing by erasing the custom input when some other input field is used, right in the component (without involving redux) because it's simpler. 

Another way would be to create additional `percentSize` field in this slice and updating it when the shortcut % buttons are clicked or even when the position size itself is changed, that would be quite usefull for the user I think, but it's more complex and requires the pecentage logic to work (and right now it kinda doesn't).